pinot-controller/src/test/java/org/apache/pinot/controller/util/JsonSerializerTest.java

```diff
@@ -0,0 +1,187 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.util;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.restlet.resources.ServerSegmentsReloadCheckResponse;
+import org.apache.pinot.common.restlet.resources.TableSegmentsReloadCheckResponse;
+import org.apache.pinot.controller.api.resources.PinotLeadControllerRestletResource;
+import org.apache.pinot.spi.utils.JsonUtils;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+
+/**
+ * Test class to check for json responses during serialization and deserialization
+ */
+public class JsonSerializerTest {
+  @Test
+  public void testTableReloadResponseSerialization()
+      throws IOException {
+    boolean needReload = true;
+    String instanceId = "instance123";
+    ServerSegmentsReloadCheckResponse response = new ServerSegmentsReloadCheckResponse(needReload, instanceId);
+    Map<String, ServerSegmentsReloadCheckResponse> serversResponse = new HashMap<>();
+    serversResponse.put(instanceId, response);
+    TableSegmentsReloadCheckResponse tableResponse = new TableSegmentsReloadCheckResponse(needReload, serversResponse);
+    String responseString = JsonUtils.objectToPrettyString(response);
+    String tableResponseString = JsonUtils.objectToPrettyString(tableResponse);
+
+    assertNotNull(responseString);
+    assertNotNull(tableResponseString);
+    JsonNode tableResponseJsonNode = JsonUtils.stringToJsonNode(tableResponseString);
+    assertTrue(tableResponseJsonNode.get("needReload").asBoolean());
+
+    JsonNode serversList = tableResponseJsonNode.get("serverToSegmentsCheckReloadList");
+    JsonNode serverResp = serversList.get("instance123");
+    assertEquals(serverResp.get("instanceId").asText(), "instance123");
+    assertTrue(serverResp.get("needReload").asBoolean());
+
+    assertEquals("{\n" + "  \"needReload\" : true,\n" + "  \"serverToSegmentsCheckReloadList\" : {\n"
```

Comment:
This is not very readable. You may disable the formatter by putting a comment `//@formatter:off` and enable it by putting `//@formatter:on` after the assert

Reply:
Refactored to a more readable one.